src/main/java/ui/PanelMenuCreator.java

```diff
@@ -0,0 +1,117 @@
+package ui;
+
+import javafx.beans.value.ChangeListener;
+import javafx.beans.value.ObservableValue;
+import javafx.geometry.Orientation;
+import javafx.scene.Node;
+import javafx.scene.control.Menu;
+import javafx.scene.control.MenuItem;
+import javafx.scene.control.ScrollBar;
+import javafx.scene.control.ScrollPane;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import ui.issuepanel.PanelControl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static ui.components.KeyboardShortcuts.CLOSE_PANEL;
+import static ui.components.KeyboardShortcuts.CREATE_LEFT_PANEL;
+import static ui.components.KeyboardShortcuts.CREATE_RIGHT_PANEL;
+
+public class PanelMenuCreator {
+    public static final Logger logger = LogManager.getLogger(MenuControl.class.getName());
+    public static final String MILESTONE_FILTER_NAME =  "milestone:curr sort:status";
+    public static final String MILESTONE_PANEL_NAME = "Current Milestone";
+
+    public static final String ASSIGNEE_FILTER_NAME =  "is:open ((is:issue assignee:me) OR (is:pr author:me))";
+    public static final String ASSIGNEE_PANEL_NAME = "Open issues and PR's";
+
+    public static final String UPDATED_FILTER_NAME = "assignee:me updated:<48";
+    public static final String UPDATED_PANEL_NAME = "Recently Updated issues";
+
+    private final PanelControl panelControl;
+    private final ScrollPane panelsScrollPane;
+
+    public PanelMenuCreator(PanelControl panelControl, ScrollPane panelsScrollPane) {
+        this.panelsScrollPane = panelsScrollPane;
+        this.panelControl = panelControl;
+    }
+
+    public Menu generatePanelMenu(){
+        Menu panelMenu = new Menu("Panels");
+        Menu autoCreatePanelMenu = new Menu("Auto-create");
+        List<MenuItem> items = new ArrayList<>();
+        List<MenuItem> autoCreateItems = new ArrayList<>();
+        items.add(createLeftPanel());
+        items.add(createRightPanel());
+        items.add(closePanel());
+        autoCreateItems.add(createCustomizedPanel(ASSIGNEE_PANEL_NAME, ASSIGNEE_FILTER_NAME));
```

Comment:
`createLeftPanel()` etc. trips me a bit when I'm reading this. Adding *Menu* or *MenuItem* suffix to menu related variables and methods could make things clearer.